Examples/Examples/All Examples/SceneKitExample.swift

```diff
@@ -0,0 +1,227 @@
+    }
+
+    public func renderingWillStart(_ metalDevice: MTLDevice, colorPixelFormat: UInt, depthStencilPixelFormat: UInt) {
+        renderer = SCNRenderer(device:metalDevice)
+        scene = SCNScene()
+        renderer.scene = scene
+
+        modelNode = SCNScene(named:"34M_17")?.rootNode;
+        scene.rootNode.addChildNode(modelNode)
+        
+        cameraNode = SCNNode();
+        let camera = SCNCamera();
+        cameraNode.camera = camera;
+        camera.usesOrthographicProjection = false;
+        scene.rootNode.addChildNode(cameraNode);
+        renderer.pointOfView = cameraNode;
+        self.setupLight()
+        // In order to use depth occlusion, align with gl-native Z handling (doesn't use reverse Z).
+        if #available(iOS 13.0, *) {
+            renderer.usesReverseZ = false
+        } else {
+            // Fallback on earlier versions, disable depth in render()
+        }
+    }
+    
+    func setupLight() {
+        // Ambient light
+        let ambientLight = SCNNode()
+        ambientLight.light = SCNLight()
+        ambientLight.light?.type = SCNLight.LightType.ambient
+        ambientLight.light?.color = UIColor(white: 0.4, alpha: 1.0)
+        modelNode.addChildNode(ambientLight)
+
+        let lightNode = SCNNode()
+        lightNode.light = SCNLight()
+        lightNode.light?.type = SCNLight.LightType.directional
+        lightNode.light?.orthographicScale = 30
+        lightNode.light?.color = UIColor(white: 0.8, alpha: 1.0)
+        lightNode.position = SCNVector3Make(-50, 100, 100);
+        lightNode.light?.zNear = 1
+        lightNode.light?.zFar = 1000
+        lightNode.light?.intensity = 2000
+        lightNode.look(at: modelNode.worldPosition)
+        modelNode.addChildNode(lightNode)
+        let pointNode = SCNNode()
+        pointNode.light = SCNLight()
+        pointNode.light?.type = SCNLight.LightType.omni
+        pointNode.light?.intensity = 3000
+        pointNode.position = SCNVector3Make(0, 25, 0)
+        modelNode.addChildNode(pointNode)
+    }
+
+    internal func makeTranslationMatrix(tx: Double, ty: Double, tz: Double) -> simd_double4x4 {
+        var matrix = matrix_identity_double4x4
+    
+        matrix[3, 0] = tx
+        matrix[3, 1] = ty
+        matrix[3, 2] = tz
+    
+        return matrix
+    }
+    
+    internal func makeScaleMatrix(xScale: Double, yScale: Double, zScale: Double) -> simd_double4x4 {
+        var matrix = matrix_identity_double4x4
+    
+        matrix[0, 0] = xScale
+        matrix[1, 1] = yScale
+        matrix[2, 2] = zScale
+    
+        return matrix
+    }
+    
+    public func render(_ parameters: CustomLayerRenderParameters, mtlCommandBuffer: MTLCommandBuffer, mtlRenderPassDescriptor: MTLRenderPassDescriptor) {
+        let m = parameters.projectionMatrix;
+        
+        // It is essential to use double precision for computation below: using simd instead
```

Comment:
Can you say why - presumably because of jittering with floats?

Reply:
Yes, mercator coordinate addresses whole world and there's just not enough precision if using 32-bit floating point computation.

Reply:
Can you add that to the comment please?